Update the code of ObstacleAvoidance.ino

flyslowly opened the pull request:
The code is made clearer and more reasonable

Related issues: #48

flyslowly committed:
Delete old version of ObstacleAvoidance.ino

This old version needs to be deleted and a new version of a file will be added

flyslowly committed:
Add new version of ObstacleAvoidance

A new version of ObstacleAvoidance.ino is added

platisd commented on ObstacleAvoidance.ino:
```diff
@@ -1,49 +1,43 @@
 
-}else{
-  car.setSpeed(100);
+ if(front.getDistance() < STOP_DISTANCE && front.getDistance() > 0){
```
This will conduct two separate measurements that can potentially yield different results.
This means that if the first call of `getDistance` returns `0` and the second one `100` your car will stop despite there not being an obstacle nearby.

flyslowly committed:
Fix problem of yield different results

Fixed from line 37 to 39, initialize a new curDistance to store the distance every time got from the sensor

platisd reviewed (approved):
Don't forget to submit this to Canvas as well!
